### PR TITLE
CAL-79 Add support for parsing the AIMIDB TRE from NITFs

### PR DESCRIPTION
**Pull request by roelens8**
#### What does this PR do?
Parses the AIMIDB TRE from NITFs. 

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@harrison-tarr 
@dcruver 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jaymcnallie
@kcwire 
#### How should this be tested?
Run a full build and make sure unit tests pass.
#### What are the relevant tickets?
CAL-79
#### Checklist:
- [ ] Documentation Updated
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by roelens8** [comment]
Rebased. The AIMIDB attribute is now a class instead of an enum.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java [hunk 1]
```diff
@@ -47,6 +47,24 @@ public static Serializable getTreValue(TreGroup tre, String key) {
         return null;
     }
 
+    public static String parseString(Tre tre, String fieldName) {
```
Comment (dcruver) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java]: This doesn't really do any parsing, so I probably would have called it 'convertToString' or something similar.

File: catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java [hunk 2]
```diff
@@ -47,6 +47,24 @@ public static Serializable getTreValue(TreGroup tre, String key) {
...
+        Serializable value = TreUtility.getTreValue(tre, fieldName);
+        if (value instanceof String) {
+            return (String) value;
+        } else {
+            return null;
+        }
+    }
+
+    public static Integer parseInteger(Tre tre, String fieldName) {
```
Comment (dcruver) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java]: convertToInteger?

Comment (roelens8) [inline catalog/imaging/imaging-transformer-nitf/src/main/java/org/codice/alliance/transformer/nitf/common/TreUtility.java]: Fixed.

**Pull request merged**